File: cc/048.cc
// Project Euler Problem 048
//
// Find the last ten digits of 1^1 + 2^2 + ... + 1000^1000.

#include <cstdio>
#include <vector>
#include "bigmath.h"

using namespace std;

int main(int argc, const char* argv[]) {
  Bignum10 sum(0);

  for (int i = 1; i <= 1000; i++) {
    Bignum10 add(1);
    for (int j=0; j<i; j++) {
      add *= i;
      add.Resize(10);
    }
    sum += add;
    sum.Resize(10);
  }
  sum.Print();
}

File: go/problems/048.go
// Project Euler Problem 048
//
// Find the last ten digits of 1^1 + 2^2 + ... + 1000^1000.

package problems

import "fmt"

func Problem048() string {
	sum := int64(0)
	for i := int64(1); i <= 1000; i++ {
		littleSum := i
		for j := int64(1); j < i; j++ {
			littleSum *= i
			littleSum %= 10000000000
		}
		sum += littleSum
		sum %= 10000000000
	}
	return fmt.Sprint(sum)
}

func init() {
	Functions[48] = Problem048
}
